state/txn.go

```diff
@@ -8,13 +8,18 @@ import (
 
 	"github.com/0xPolygon/polygon-edge/chain"
 	"github.com/0xPolygon/polygon-edge/crypto"
-	"github.com/0xPolygon/polygon-edge/helper/keccak"
 	"github.com/0xPolygon/polygon-edge/state/runtime"
 	"github.com/0xPolygon/polygon-edge/types"
 )
 
 var emptyStateHash = types.StringToHash("0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421")
 
+type readSnapshot interface {
+	GetStorage(addr types.Address, root types.Hash, key types.Hash) types.Hash
```

Comment:
What is root there?

Reply:
`root` is the Storage Root for `addr`. In the future, The `readSnapshot` will know from the `addr` automatically which is the root.